setup.cfg

```diff
@@ -31,10 +31,10 @@ packages = find:
 include_package_data = True
 python_requires = >=3.6
 install_requires =
-    pyglet
-    pillow
-    numpy
-    pymunk
+    pyglet>=1.5.8,<2
+    pillow == 8.0.1
```

Comment:
Is it worth doing ranges on these other dependencies instead of pinning them to specific versions?  

As written, if an app depends on arcade AND other modules... there is a chance that the other module also has pillow, numpy, or pymunk in its dependency tree.  This means version conflicts could arise.  For instance, if an app imports arcade and module "X", and module "X" has numpy 1.18.1 as a dependency... there is a conflict because both numpy 1.18.1 (from module "X") and numpy 1.19.4 (from arcade) can not both be installed.  So, my understanding is that libraries (such as arcade) should use ranged dependencies where possible.

Now, this would probably require testing to find out what version ranges of pillow, numpy and pymunk would work with arcade.  Semantic versioning could help here, as API breaking changes are usually indicated by bumping the major version number (I don't know if numpy, pillow, pymunk follow this. convension...)

One (of many) references: https://www.promptworks.com/blog/pin-all-dependencies/ (see the "Instance #1: Your software is a third-party module" under "When should you not pin a dependency?").

Reply:
That's a good catch. I'll do some research into this tonight, pinning specific versions is probably the right way to go if that packages dependencies are pinned. I will go over everything and find out specific versions and if they should be ranges or not and submit another commit to handle that if necessary.

Reply:
This issue is resolved in https://github.com/pythonarcade/arcade/pull/780/commits/7d7c87d687af13669f90f320f4243cb022b417fa There are no cross dependencies for anything here, so I've pinned Pyglet to ensure it doesn't install version 2 if that releases and we're unprepared for it.

pytiled_parser is pinned at the exact version in order to ensure compatibility if we release version 1.0.0 of that before the Arcade release is ready for that.

Everything else is pinned to approximate minor versions, so that it will install the latest patch version of the most recent compatible minor version.